lib/apiservers/engine/backends/container.go

```diff
@@ -1605,6 +1599,70 @@ func (c *Container) containerAttach(name string, ca *backend.ContainerAttachConf
 	return nil
 }
 
+// ContainerRename changes the name of a container, using the oldName
+// to find the container. An error is returned if newName is already
+// reserved.
+func (c *Container) ContainerRename(oldName, newName string) error {
+	defer trace.End(trace.Begin(newName))
+
+	if oldName == "" || newName == "" {
+		err := fmt.Errorf("neither old nor new names may be empty")
+		log.Errorf("%s", err.Error())
+		return err
```

Comment:
To echo @hmahmood's comment - we should return docker errors with appropriate response codes. See https://docs.docker.com/engine/api/v1.25/#operation/ContainerRename.

Also, I don't think this error is worth logging.